enterprise/frontend/src/metabase-enterprise/metabot/components/MetabotChat/MetabotChat.tsx

```diff
@@ -1,147 +1,282 @@
+  ActionIcon,
+  Box,
+  type BoxProps,
+  Button,
+  Flex,
+  Icon,
+  Loader,
+  Paper,
+  Stack,
+  Text,
+  Textarea,
+  UnstyledButton,
+} from "metabase/ui";
 
 import { useMetabotAgent } from "../../hooks";
-import { MetabotIcon } from "../MetabotIcon";
+import { AIMarkdown } from "../AIMarkdown/AIMarkdown";
 
 import Styles from "./MetabotChat.module.css";
-import { useAutoCloseMetabot } from "./useAutoCloseMetabot";
+import { useAutoscrollMessages } from "./hooks";
 
-const MIN_INPUT_HEIGHT = 42;
-const ANIMATION_DURATION_MS = 300;
-
-export const MetabotChat = ({ onClose }: { onClose: () => void }) => {
+export const MetabotChat = () => {
   const textareaRef = useRef<HTMLTextAreaElement>(null);
+  const messagesRef = useRef<HTMLDivElement>(null);
+  const headerRef = useRef<HTMLDivElement>(null);
 
   const [input, setMessage] = useState("");
 
   const metabot = useMetabotAgent();
 
-  const resetInput = useCallback(() => {
-    setMessage("");
-    setInputExpanded(false);
-  }, []);
+  useAutoscrollMessages(headerRef, messagesRef, metabot.messages);
+
+  const hasMessages = metabot.messages.length > 0;
+  const suggestedPrompts = useMemo(() => {
+    const prompts = metabot.suggestedPrompts.data?.prompts ?? [];
+    return _.shuffle(prompts).slice(0, 3);
+  }, [metabot.suggestedPrompts]);
+  const hasSuggestions = suggestedPrompts.length > 0;
+
+  const handleSubmitInput = (input: string) => {
+    if (metabot.isDoingScience) {
+      return;
+    }
 
-  const handleSend = () => {
     const trimmedInput = input.trim();
     if (!trimmedInput.length || metabot.isDoingScience) {
       return;
     }
-    resetInput();
+    setMessage("");
     metabot
       .submitInput(trimmedInput)
       .catch((err) => console.error(err))
       .finally(() => textareaRef.current?.focus());
   };
 
+  const { setVisible } = metabot;
   const handleClose = useCallback(() => {
-    resetInput();
-    onClose();
-  }, [resetInput, onClose]);
-
-  useAutoCloseMetabot({
-    hasUserInput: !!input,
-  });
-
-  const [inputExpanded, setInputExpanded] = useState(false);
-  const handleMaybeExpandInput = () => {
-    const textarea = textareaRef.current;
-    if (!textarea) {
-      return;
-    }
-
-    const isMultiRow = textarea.scrollHeight > MIN_INPUT_HEIGHT;
-    if (inputExpanded !== isMultiRow) {
-      setInputExpanded(isMultiRow);
-    }
-    // keep scrolled to bottom
-    textarea.scrollTop = Math.max(MIN_INPUT_HEIGHT, textarea.scrollHeight);
-  };
+    setMessage("");
+    setVisible(false);
+  }, [setVisible]);
 
   const handleInputChange = (value: string) => {
     setMessage(value);
   };
 
-  const inputPlaceholder = t`Tell me to do something, or ask a question`;
-  const placeholder = metabot.isDoingScience
-    ? t`Doing science...`
-    : inputPlaceholder;
-
   return (
-    <Box
-      className={Styles.container}
-      style={{ animationDuration: `${ANIMATION_DURATION_MS}ms` }}
-      data-testid="metabot-chat"
+    <Sidebar
```

Comment:
There's a lot of markup in this component. I'm currently working on splitting this up, but you can imagine components where I have comments for now.